internal/app/api/v1/executor.go

```diff
@@ -113,19 +127,42 @@ func mapExecutorCRDToExecutorDetails(item executorv1.Executor) testkube.Executor
 	}
 }
 
-func mapExecutorCreateRequestToExecutorCRD(request testkube.ExecutorCreateRequest) executorv1.Executor {
-	return executorv1.Executor{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      request.Name,
-			Namespace: request.Namespace,
-			Labels:    request.Labels,
-		},
-		Spec: executorv1.ExecutorSpec{
-			ExecutorType: request.ExecutorType,
-			Types:        request.Types,
-			URI:          request.Uri,
-			Image:        request.Image,
-			JobTemplate:  request.JobTemplate,
-		},
+func (s TestkubeAPI) getExecutorCRD(c *fiber.Ctx, executor testkube.ExecutorCreateRequest) error {
+	if executor.JobTemplate != "" {
+		executor.JobTemplate = fmt.Sprintf("%q", executor.JobTemplate)
+	}
+
+	data, err := crd.ExecuteTemplate(crd.TemplateExecutor, executor)
+	if err != nil {
+		return s.Error(c, http.StatusBadRequest, err)
 	}
+
+	c.Context().SetContentType(mediaTypeYAML)
+	return c.SendString(data)
+}
+
+func (s TestkubeAPI) getExecutorCRDs(c *fiber.Ctx, executors []testkube.ExecutorCreateRequest) error {
+	data := ""
+	firstEntry := true
```

Comment:
Looks very similar to `common.UIPrintCRD` logic

Reply:
refactored